libs/platform/src/lib/shared/domain/base-data-provider.ts

```diff
@@ -67,6 +67,9 @@ export class BaseDataProvider<T> extends DataProvider<T> {
             value = value.call(item);
         } else if (isJsObject(value)) {
             return this.hasObjectValue(item, pattern);
+        } else if (this._matchingStrategy === MatchingStrategy.STARTS_WITH_PER_TERM) {
+            const reqexp = new RegExp(`(\\s|^)${pattern}`, 'gi');
```

Comment:
I saw this regexp 3 times so let's create a function where we will be testing for matching to this regexp.